<div style="max-width: 700px; margin: 20px auto 30px; padding: 20px; text-align: center; border-bottom: 2px solid #e5e7eb;">
  <a href="1_1_simul_univariate.ipynb" style="display: inline-block; padding: 10px 24px; background: #f9fafb; color: #38549c; text-decoration: none; border-radius: 8px; font-weight: 600; font-size: 14px; border: 1px solid #e5e7eb; margin-right: 12px;">
    ← Part 1 - Univariate Time Series Model Simulations
  </a>
</div>

<div style="max-width: 880px; margin: 20px auto 22px; padding: 0px; border-radius: 18px; border: 1px solid #e5e7eb; background: linear-gradient(180deg, #ffffff 0%, #f9fafb 100%); box-shadow: 0 8px 26px rgba(0,0,0,0.06); overflow: hidden;">
  <div style="padding: 34px 32px 14px; text-align: center; line-height: 1.38;">
    <div style="font-size: 13px; letter-spacing: 0.14em; text-transform: uppercase; color: #6b7280; font-weight: bold; margin-bottom: 5px;">
      Session #1 • Part 2
    </div>
    <!-- <div style="font-size: 29px; font-weight: 800; color: #14276c; margin-bottom: 4px;">
      Linear Time Series Models
    </div> -->
    <div style="font-size: 29px; font-weight: 800; color: #14276c; margin-bottom: 4px;">
      ARMA Model Estimation and Selection
    </div>
    <div style="font-size: 16.5px; color: #374151; font-style: italic; margin-bottom: 0;">
      Advanced Training School: Methods for Time Series
    </div>
  </div>
  <div style="background: none; text-align: center; margin: 30px 0 10px;">
    <img src="https://www.cemfi.es/images/Logo-Azul.png" alt="CEMFI Logo" style="width: 158px; filter: drop-shadow(0 2px 12px rgba(56,84,156,0.05)); margin-bottom: 0;">
  </div>
  <div style="font-family: 'Times New Roman', Times, serif; color: #38549c; text-align: center; font-size: 1.22em; font-weight: bold; margin-bottom: 0px;">
    Jesus Villota Miranda © 2025
  </div>
  <div style="font-family: 'Times New Roman', Times, serif; color: #38549c; text-align: center; font-size: 1em; margin-top: 7px; margin-bottom: 20px;">
    <a href="mailto:jesus.villota@cemfi.edu.es" style="color: #38549c; text-decoration: none; margin-right:8px;">jesus.villota@cemfi.edu.es</a>
    <span style="color:#9fa7bd;">|</span>
    <a href="https://www.linkedin.com/in/jesusvillotamiranda/" target="_blank" style="color: #38549c; text-decoration: none; margin-left:7px;">LinkedIn</a>
  </div>
</div>

<div style="max-width: 700px; margin: 20px auto 30px; padding: 24px 32px; border-radius: 18px; border: 1px solid #e5e7eb; background: linear-gradient(180deg, #ffffff 0%, #f9fafb 100%); box-shadow: 0 8px 26px rgba(0,0,0,0.06);">
<!-- <div style="font-size: 14px; letter-spacing: 0.1em; text-transform: uppercase; color: #6b7280; font-weight: bold; margin-bottom: 12px; text-align: center;">Session #1 • Part II</div> -->
<div style="font-size: 22px; font-weight: 700; color: #14276c; margin-bottom: 16px; text-align: center;">Table of Contents</div><div style="font-size: 15px; color: #374151; line-height: 1.8;"><div style="margin-bottom: 12px;"><strong>1.</strong> <a href="#section-1" style="color: #38549c; text-decoration: none;">Introduction to ARMA Models</a></div><div style="margin-bottom: 12px;"><strong>2.</strong> <a href="#section-2" style="color: #38549c; text-decoration: none;"> Modeling the VIX Index</a></div><div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">2.1 VIX Historical Evolution and Distribution</div><div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">2.2 Log-VIX Transformation</div><div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">2.3 Unit Root Testing</div><div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">2.4 ARMA Model Selection (AIC/BIC)</div><div style="margin-left: 20px; margin-bottom: 8px; font-size: 14px;">2.5 ARMA(2,1) Estimation</div><div style="margin-left: 20px; margin-bottom: 0px; font-size: 14px;">2.6 Residual Diagnostics</div></div></div>

<link rel="stylesheet" href="notebook_styles.css">

<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-1"></a>
<h2 class="styled-header">1) Introduction to ARMA Models</h2>

**ARMA(p,q)** models combine autoregressive (AR) and moving average (MA) components to capture both the persistence and shock dynamics in time series data:

$$
Y_t = \phi_1 Y_{t-1} + \cdots + \phi_p Y_{t-p} + \varepsilon_t + \theta_1 \varepsilon_{t-1} + \cdots + \theta_q \varepsilon_{t-q}
$$

where:
- $Y_t$ is the time series at time $t$
- $\phi_1, \ldots, \phi_p$ are the **autoregressive coefficients** (capture persistence from past values)
- $\theta_1, \ldots, \theta_q$ are the **moving average coefficients** (capture impact of past shocks)
- $\varepsilon_t$ is white noise with $E[\varepsilon_t] = 0$ and $\text{Var}(\varepsilon_t) = \sigma^2$

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Why ARMA Models?**

ARMA models provide a **parsimonious** way to represent complex autocorrelation patterns:
- **AR component**: Captures direct dependence on past values (long memory)
- **MA component**: Captures dependence on past forecast errors (smoothing)
- **Combined**: Can approximate a wide range of stationary processes with fewer parameters than pure AR or MA models

</div>

#### Model Selection: Information Criteria

To determine optimal lag orders $(p, q)$, we use **information criteria** that balance model fit against complexity:

**Akaike Information Criterion (AIC)**:
$$
\text{AIC} = -2\ln(L) + 2k
$$

**Bayesian Information Criterion (BIC)**:
$$
\text{BIC} = -2\ln(L) + k\ln(T)
$$

where:
- $L$ = maximized likelihood
- $k$ = number of estimated parameters
- $T$ = sample size

<div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 18px; border-radius: 8px; margin: 20px 0;">

**⚠️ Important: Lower Values are Better**

Both criteria penalize model complexity, but **BIC** penalizes more heavily:
- **AIC**: Tends to select larger models (better for prediction)
- **BIC**: Tends to select smaller models (better for parsimony and interpretation)
- **Strategy**: Choose model with *minimum* AIC or BIC, considering both criteria

</div>

#### Key Properties for Stationarity

For an ARMA process to be **stationary**, the roots of the AR characteristic polynomial must lie outside the unit circle:

$$
1 - \phi_1 z - \phi_2 z^2 - \cdots - \phi_p z^p = 0 \quad \text{requires} \quad |z| > 1
$$

**Practical Check**: For ARMA(2,1), stationarity requires $|\phi_1 + \phi_2| < 1$, $\phi_2 - \phi_1 < 1$, and $-1 < \phi_2 < 1$.

<link rel="stylesheet" href="../notebook_styles.css">

<a id="section-2"></a>
<h2 class="styled-header">2) Modeling of the VIX Index</h2>

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**The VIX Challenge**

The **VIX** (CBOE Volatility Index) measures market expectations of near-term volatility conveyed by S&P 500 index option prices. Modeling VIX presents unique challenges:
- **Non-normal distribution**: Heavy right tail (volatility spikes)
- **Mean reversion**: Volatility tends to return to long-run average
- **Persistent autocorrelation**: High volatility tends to cluster

</div>

#### Objectives

This analysis demonstrates the complete ARMA modeling workflow:

1. **Explore** VIX distributional properties and temporal patterns
2. **Transform** data to achieve approximate normality
3. **Test** for stationarity (prerequisite for ARMA modeling)
4. **Select** optimal ARMA specification using information criteria
5. **Estimate** the chosen model and interpret coefficients
6. **Diagnose** residuals to validate model adequacy
7. **Forecast** out-of-sample and evaluate predictive performance

### 2.1 VIX Historical Evolution and Distribution

#### Initial Data Exploration

We begin by loading daily VIX data spanning **January 1990 to November 2025**, examining its temporal evolution and distributional characteristics.

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 16px; border-radius: 6px; margin: 16px 0;">

**What to Look For:**
- **Level shifts**: Major crisis periods (2000-02 dot-com, 2008-09 financial crisis)
- **Volatility clustering**: High-volatility episodes tend to persist
- **Distributional shape**: Skewness and kurtosis indicate departure from normality

</div>

In [32]:
* Load VIX data from processed file
global processed_data "../../data/processed"  // Define path to processed data directory
use "$processed_data/vix_daily.dta", clear  // Load VIX daily dataset and clear memory

* Display first and last observations
list date vix in 1/5
list date vix in -5/-1

* Display summary information
describe
summarize vix

* Summary statistics for VIX
quietly summarize vix, detail
display "VIX Summary Statistics:"
display "Mean:     " %9.4f r(mean)
display "Std Dev:  " %9.4f r(sd)
display "Skewness: " %9.4f r(skewness)
display "Kurtosis: " %9.4f r(kurtosis)
display "Min:      " %9.4f r(min)
display "Max:      " %9.4f r(max)

* Plot VIX historical evolution
tsline vix, ///
    title("VIX Index - Historical Evolution", size(medium)) ///
    subtitle("Jan 1990 to Jan 2015") ///
    ytitle("VIX Index") ///
    xtitle("Date") ///
    tlabel(, format(%tdCY)) ///
    scheme(s2color)
/* graph export "../../output/vix_evolution.png", replace width(1200) */

* Kernel density estimate for VIX vs normal density
kdensity vix, ///
    normal ///
    title("VIX - Kernel Density vs Normal Distribution", size(medium)) ///
    xtitle("VIX Index") ///
    ytitle("Density") ///
    legend(label(1 "Kernel density") label(2 "Normal density")) ///
    scheme(s2color)
/* graph export "../../output/vix_kde.png", replace width(1200) */





     +-------------------+
     |      date     vix |
     |-------------------|
  1. | 02jan1990   17.24 |
  2. | 03jan1990   18.19 |
  3. | 04jan1990   19.22 |
  4. | 05jan1990   20.11 |
  5. | 08jan1990   20.26 |
     +-------------------+


      +-------------------+
      |      date     vix |
      |-------------------|
9055. | 07nov2025   19.08 |
9056. | 10nov2025    17.6 |
9057. | 11nov2025   17.28 |
9058. | 12nov2025   17.51 |
9059. | 13nov2025      20 |
      +-------------------+


Contains data from ../../data/processed/vix_daily.dta
 Observations:         9,059                  
    Variables:             3                  21 Nov 2025 00:40
--------------------------------------------------------------------------------
Variable      Storage   Display    Value
    name         type    format    label      Variable label
--------------------------------------------------------------------------------
vix             float   %9.0g                 VIXCLS
date           

### 2.2 Log-VIX Transformation

#### Why Log Transformation?

<div style="background: linear-gradient(135deg, #fef3c7 0%, #fde68a 100%); border-left: 4px solid #f59e0b; padding: 18px; border-radius: 8px; margin: 20px 0;">

**⚠️ The Problem with Raw VIX**

Raw VIX exhibits:
- **Positive skewness**: Long right tail from crisis episodes
- **Excess kurtosis**: Fat tails beyond normal distribution
- **Heteroskedasticity**: Variance changes over time

These properties violate ARMA assumptions requiring **approximate normality** and **constant variance**.

</div>

#### The Log Transformation Solution

The natural logarithm transformation addresses these issues:

$$
\text{Log-VIX}_t = \ln(\text{VIX}_t)
$$

**Benefits**:
1. **Variance stabilization**: Reduces heteroskedasticity by compressing large values
2. **Symmetry improvement**: Reduces right skewness toward normality
3. **Interpretability**: Changes in log-VIX approximate *percentage changes* in VIX

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Key Takeaway**: Log transformation is standard practice when modeling financial volatility indices, making the data more suitable for linear time series models like ARMA.

</div>

In [33]:
* Create log-VIX transformation
/* generate lgvix = ln(vix)  // Create natural log transformation of VIX */
label variable lgvix "Log of VIX index"  // Assign descriptive label to new variable

* Summary statistics for log-VIX
quietly summarize lgvix, detail  // Compute detailed summary statistics without output
display "Log-VIX Summary Statistics:"
display "Mean:     " %9.4f r(mean)
display "Std Dev:  " %9.4f r(sd)
display "Skewness: " %9.4f r(skewness)
display "Kurtosis: " %9.4f r(kurtosis)
display "Min:      " %9.4f r(min)
display "Max:      " %9.4f r(max)




Log-VIX Summary Statistics:

Mean:        2.9048

Std Dev:     0.3426

Skewness:    0.6674

Kurtosis:    3.4865

Min:         2.2127

Max:         4.4151


In [34]:
* Kernel density estimate for log-VIX vs normal density
kdensity lgvix, ///
    normal ///
    title("Log-VIX - Kernel Density vs Normal Distribution", size(medium)) ///
    xtitle("Log-VIX") ///
    ytitle("Density") ///
    legend(label(1 "Kernel density") label(2 "Normal density")) ///
    scheme(s2color)
/* graph export "../../output/logvix_kde.png", replace width(1200) */


file /Users/jesusvillotamiranda/.stata_kernel_cache/graph18.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph18.pdf saved as PDF
    format


In [35]:
* ACF for log-VIX
ac lgvix, ///
    lags(40) ///
    title("Log-VIX - Autocorrelation Function", size(medium)) ///
    ytitle("Autocorrelation") ///
    xtitle("Lag") ///
    scheme(s2color)
/* graph export "../../output/logvix_acf.png", replace width(1200) */

(note: time series has 1949 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph19.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph19.pdf saved as PDF
    format


In [36]:
* PACF for log-VIX
pac lgvix, ///
    lags(40) ///
    title("Log-VIX - Partial Autocorrelation Function", size(medium)) ///
    ytitle("Partial Autocorrelation") ///
    xtitle("Lag") ///
    scheme(s2color)
/* graph export "../../output/logvix_pacf.png", replace width(1200) */

(note: time series has 1949 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph20.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph20.pdf saved as PDF
    format


### 2.3 Unit Root Testing for Stationarity

#### Why Stationarity Matters

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Critical Requirement**

ARMA models are **only valid** for **stationary** processes. A non-stationary series:
- Has time-varying mean and/or variance
- Leads to spurious regression results
- Violates asymptotic distribution theory for inference

</div>

#### The DF-GLS Test

We employ the **DF-GLS (Dickey-Fuller GLS)** test, which is more powerful than the standard ADF test, especially for series with persistent autocorrelation.

**Hypotheses**:
$$
\begin{aligned}
H_0 &: \text{Series has a unit root (non-stationary)} \\
H_1 &: \text{Series is stationary}
\end{aligned}
$$

**Decision Rule**:
- If test statistic $<$ critical value → **Reject $H_0$** → Series is stationary
- If test statistic $\geq$ critical value → **Fail to reject $H_0$** → Series may be non-stationary

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Expected Result for Log-VIX**: Given the mean-reverting nature of volatility, we anticipate **rejecting the unit root hypothesis**, confirming log-VIX is stationary and suitable for ARMA modeling.

</div>

In [37]:
* DF-GLS unit root test for log-VIX
* Create continuous series for dfgls (requires no gaps)
preserve  // Save current dataset state to restore later
drop if missing(lgvix)  // Remove observations with missing log-VIX values
gen t = _n  // Generate continuous time index from 1 to N
tsset t  // Declare dataset as time series with variable t as time index
dfgls lgvix, maxlag(10) notrend  // Run DF-GLS test with max 10 lags and no trend
display "H0: log-VIX has a unit root (non-stationary)"
display "If test statistic < critical value, reject H0 (series is stationary)"
restore  // Restore dataset to state before preserve



(0 observations deleted)



Time variable: t, 1 to 9059
        Delta: 1 unit


DF-GLS test for unit root                Number of obs = 9,048
Variable: lgvix
Lag selection: User specified            Maximum lag   =    10

                             -------- Critical value ---------
 [lags]       DF-GLS mu           1%           5%          10%
--------------------------------------------------------------
     10          -6.499       -2.580       -1.949       -1.625
      9          -6.394       -2.580       -1.949       -1.625
      8          -6.500       -2.580       -1.949       -1.626
      7          -6.761       -2.580       -1.949       -1.626
      6          -7.030       -2.580       -1.949       -1.626
      5          -7.336       -2.580       -1.950       -1.626
      4          -7.527       -2.580       -1.950       -1.626
      3          -7.886       -2.580       -1.950       -1.626
      2          -8.229       -2.580       -1.950       -1.626
      1          -8

#### Interpretation of Unit Root Results

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 What the Results Reveal:**

1. **Stationarity Confirmation**: Rejection of the unit root null hypothesis validates that log-VIX fluctuates around a stable mean with bounded variance

2. **ARMA Validity**: Stationarity ensures:
   - Autocorrelations decay to zero (necessary for ARMA identification)
   - Maximum likelihood estimation is consistent
   - Standard errors and hypothesis tests are asymptotically valid

3. **Economic Interpretation**: Volatility is **mean-reverting**—extreme levels (both high and low) tend to return to long-run average, a key property exploited by volatility traders

</div>

### 2.4 ARMA Model Selection Using Information Criteria

#### The Model Selection Problem

With confirmed stationarity, we face the challenge: **Which ARMA(p,q) specification best represents the data?**

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**The Trade-off**

- **Too few lags**: Model is **underfit** → fails to capture all autocorrelation → biased estimates
- **Too many lags**: Model is **overfit** → captures noise as signal → inefficient estimates, poor forecasts

</div>

#### Systematic Search Strategy

We estimate a grid of candidate models and compare them using AIC and BIC:

**Candidate Set**:
- Pure AR models: ARMA(1,0) through ARMA(6,0)
- Pure MA models: ARMA(0,1) through ARMA(0,3)
- Mixed ARMA: ARMA(1,1) and ARMA(2,1)

**Selection Criterion**:
$$
\text{Optimal model} = \arg\min_{(p,q)} \text{AIC}(p,q) \quad \text{or} \quad \arg\min_{(p,q)} \text{BIC}(p,q)
$$

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Expected Pattern**: For financial volatility series with **long memory** (persistent autocorrelation) and **transient shocks**, we typically expect:
- Low-order AR terms (capture persistence)
- Low-order MA terms (capture shock absorption)

</div>

In [38]:
* ARMA Model Selection - Compare different specifications
* We'll estimate AR(1) through AR(6), MA(1) through MA(3), and mixed ARMA models

display "ARMA Model Selection for Log-VIX"
display "=================================="
display ""

* Create a matrix to store results
matrix results = J(11, 4, .)  // Initialize 11x4 matrix with missing values
matrix colnames results = "p" "q" "AIC" "BIC"  // Name columns for AR order, MA order, AIC, and BIC

local row = 1  // Initialize row counter for matrix storage

* AR models: ARMA(p,0) for p=1 to 6
forvalues p = 1/6 {  // Loop through AR orders from 1 to 6
    quietly arima lgvix, ar(1/`p') vce(robust)  // Estimate AR(p) model with robust standard errors
    estat ic  // Compute information criteria
    matrix temp = r(S)  // Store results matrix
    matrix results[`row', 1] = `p'  // Store AR order
    matrix results[`row', 2] = 0  // Store MA order (zero for pure AR)
    matrix results[`row', 3] = temp[1,5]  // Extract and store AIC
    matrix results[`row', 4] = temp[1,6]  // Extract and store BIC
    local row = `row' + 1  // Increment row counter
}

* MA models: ARMA(0,q) for q=1 to 3
forvalues q = 1/3 {  // Loop through MA orders from 1 to 3
    quietly arima lgvix, ma(1/`q') vce(robust)  // Estimate MA(q) model with robust standard errors
    estat ic  // Compute information criteria
    matrix temp = r(S)  // Store results matrix
    matrix results[`row', 1] = 0  // Store AR order (zero for pure MA)
    matrix results[`row', 2] = `q'  // Store MA order
    matrix results[`row', 3] = temp[1,5]  // Extract and store AIC
    matrix results[`row', 4] = temp[1,6]  // Extract and store BIC
    local row = `row' + 1  // Increment row counter
}

* ARMA(1,1)
quietly arima lgvix, ar(1) ma(1) vce(robust)  // Estimate ARMA(1,1) model with robust standard errors
estat ic  // Compute information criteria
matrix temp = r(S)  // Store results matrix
matrix results[`row', 1] = 1  // Store AR order
matrix results[`row', 2] = 1  // Store MA order
matrix results[`row', 3] = temp[1,5]  // Extract and store AIC
matrix results[`row', 4] = temp[1,6]  // Extract and store BIC
local row = `row' + 1  // Increment row counter

* ARMA(2,1)
quietly arima lgvix, ar(1/2) ma(1) vce(robust)  // Estimate ARMA(2,1) model with robust standard errors
estat ic  // Compute information criteria
matrix temp = r(S)  // Store results matrix
matrix results[`row', 1] = 2  // Store AR order
matrix results[`row', 2] = 1  // Store MA order
matrix results[`row', 3] = temp[1,5]  // Extract and store AIC
matrix results[`row', 4] = temp[1,6]  // Extract and store BIC

* Display results
display ""
display "Model Selection Results:"
display "------------------------"
matrix list results, format(%9.4f)


ARMA Model Selection for Log-VIX








Akaike's information criterion and Bayesian information criterion

-----------------------------------------------------------------------------
       Model |          N   ll(null)  ll(model)      df        AIC        BIC
-------------+---------------------------------------------------------------
           . |      9,059          .   11381.45       3   -22756.9  -22735.57
-----------------------------------------------------------------------------
Note: BIC uses N = number of observations. See [R] BIC note.

Akaike's information criterion and Bayesian information criterion

-----------------------------------------------------------------------------
       Model |          N   ll(null)  ll(model)      df        AIC        BIC
-------------+---------------------------------------------------------------
           . |      9,059          .   11450.28       4  -22892.57  -22864.12
------------------------------------------------------------

### 2.5 ARMA(2,1) Estimation and Interpretation

#### The Selected Model

Based on information criteria, we estimate the **ARMA(2,1)** specification:

$$
\text{lgvix}_t = \phi_1 \cdot \text{lgvix}_{t-1} + \phi_2 \cdot \text{lgvix}_{t-2} + \varepsilon_t + \theta_1 \cdot \varepsilon_{t-1}
$$

**Estimation Method**: Maximum likelihood with **robust standard errors** (accounts for potential heteroskedasticity and autocorrelation in residuals)

#### Understanding the Components

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Coefficient Interpretation**

1. **$\phi_1$ (AR1 coefficient)**: Direct effect of yesterday's log-VIX on today
   - Positive $\phi_1$ → persistence (high volatility begets high volatility)
   - Magnitude indicates strength of one-period memory

2. **$\phi_2$ (AR2 coefficient)**: Direct effect of two-day-ago log-VIX on today
   - Controls for second-order dynamics
   - Often smaller in magnitude than $\phi_1$

3. **$\theta_1$ (MA1 coefficient)**: Effect of yesterday's forecast error (shock)
   - Captures how quickly market absorbs new information
   - Negative $\theta_1$ → overshooting correction

4. **Overall Persistence**: $\phi_1 + \phi_2$ measures long-run memory
   - If $\phi_1 + \phi_2 \approx 1$ → high persistence (slow mean reversion)
   - If $\phi_1 + \phi_2 < 1$ → stationary (required condition satisfied)

</div>

#### Model Quality Indicators

After estimation, check:
- **Statistical significance**: $p$-values $< 0.05$ for all coefficients
- **Information criteria**: Confirm AIC/BIC are among the lowest
- **Stationarity condition**: Verify $|\phi_1 + \phi_2| < 1$

In [39]:
* Estimate ARMA(2,1) model for log-VIX
arima lgvix, ar(1/2) ma(1) vce(robust)  // Estimate ARMA(2,1) with AR(1), AR(2), and MA(1) terms using robust SE

* Display full estimation results
estat ic  // Display information criteria for model comparison


(note:  insufficient memory or observations to estimate usual
starting values [2])

Number of gaps in sample = 1949
(note: filtering over missing observations)

(setting optimization to BHHH)
Iteration 0:   log pseudolikelihood = -3148.9011  (not concave)
Iteration 1:   log pseudolikelihood =  4513.0107  
Iteration 2:   log pseudolikelihood =  5600.0015  
Iteration 3:   log pseudolikelihood =  6107.6875  
Iteration 4:   log pseudolikelihood =  10529.408  
(switching optimization to BFGS)
Iteration 5:   log pseudolikelihood =  10763.231  
Iteration 6:   log pseudolikelihood =  11352.161  
Iteration 7:   log pseudolikelihood =  11371.627  
Iteration 8:   log pseudolikelihood =  11386.643  
Iteration 9:   log pseudolikelihood =  11387.939  
Iteration 10:  log pseudolikelihood =  11388.709  
Iteration 11:  log pseudolikelihood =  11389.498  
Iteration 12:  log pseudolikelihood =  11401.214  
Iteration 13:  log pseudolikelihood =  11418.615  
Iteration 14:  log pseudolikelihood =  11443.36

In [40]:
* Interpret ARMA(2,1) coefficients
display ""
display "ARMA(2,1) Model Interpretation:"
display "================================"
display "Model: lgvix_t = φ₁·lgvix_{t-1} + φ₂·lgvix_{t-2} + ε_t + θ₁·ε_{t-1}"
display ""
display "AR(1) coefficient (φ₁): Shows effect of previous period's log-VIX"
display "AR(2) coefficient (φ₂): Shows effect of two-period lagged log-VIX"
display "MA(1) coefficient (θ₁): Shows effect of previous period's shock"
display ""
display "Persistence: Sum of AR coefficients indicates overall persistence"
display "If |φ₁ + φ₂| < 1, the process is stationary"




ARMA(2,1) Model Interpretation:


Model: lgvix_t = φ₁·lgvix_{t-1} + φ₂·lgvix_{t-2} + ε_t + θ₁·ε_{t-1}



AR(1) coefficient (φ₁): Shows effect of previous period's log-VIX

AR(2) coefficient (φ₂): Shows effect of two-period lagged log-VIX

MA(1) coefficient (θ₁): Shows effect of previous period's shock



Persistence: Sum of AR coefficients indicates overall persistence

If |φ₁ + φ₂| < 1, the process is stationary


### 2.6 Residual Diagnostics: Validating Model Adequacy

#### The Crucial Test

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Why Residual Checks Matter**

A well-specified ARMA model should capture **all** systematic patterns in the data. If the model is correct, residuals $\hat{\varepsilon}_t$ should be **white noise**:

$$
\hat{\varepsilon}_t \sim \text{WN}(0, \sigma^2) \quad \Rightarrow \quad \begin{cases}
E[\hat{\varepsilon}_t] = 0 \\
\text{Var}(\hat{\varepsilon}_t) = \sigma^2 \\
\text{Cov}(\hat{\varepsilon}_t, \hat{\varepsilon}_s) = 0 \quad \forall t \neq s
\end{cases}
$$

**If residuals are NOT white noise** → model is misspecified → inference is invalid

</div>

#### Diagnostic Tools

**1. ACF and PACF of Residuals**
- Plot autocorrelations at multiple lags
- *Ideal outcome*: All spikes within confidence bands
- *Problem signs*: Significant spikes → remaining autocorrelation

**2. Ljung-Box Q-Test**
- Formal test for joint significance of autocorrelations
- $H_0$: No autocorrelation up to lag $K$
- $H_1$: At least one autocorrelation is non-zero

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Success Criteria**: 
- ACF/PACF show **no significant spikes**
- Ljung-Box test yields **$p$-value $> 0.05$** → fail to reject white noise hypothesis
- This validates that ARMA(2,1) has successfully captured all autocorrelation structure

</div>

In [41]:
* Get residuals from ARMA(2,1) model
predict resid_arma21, residuals  // Generate residuals from last estimated ARMA(2,1) model

* ACF of residuals
ac resid_arma21, ///
    lags(40) ///
    title("ARMA(2,1) Residuals - ACF", size(medium)) ///
    ytitle("Autocorrelation") ///
    xtitle("Lag") ///
    scheme(s2color)
/* graph export "../../output/arma21_resid_acf.png", replace width(1200) */



(note: time series has 1949 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph21.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph21.pdf saved as PDF
    format


In [42]:
* PACF of residuals
pac resid_arma21, ///
    lags(40) ///
    title("ARMA(2,1) Residuals - PACF", size(medium)) ///
    ytitle("Partial Autocorrelation") ///
    xtitle("Lag") ///
    scheme(s2color)
/* graph export "../../output/arma21_resid_pacf.png", replace width(1200) */

(note: time series has 1949 gaps)

file /Users/jesusvillotamiranda/.stata_kernel_cache/graph22.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph22.pdf saved as PDF
    format


In [43]:
* Ljung-Box Q-test for residual autocorrelation
wntestq resid_arma21, lags(20)  // Test for white noise in residuals up to lag 20
display ""
display "H0: No autocorrelation in residuals up to lag 20"
display ""
display "Interpretation:"
display "- If p-value > 0.05, we fail to reject H0 (residuals are white noise)"
display "- If p-value <= 0.05, we reject H0 (residuals show autocorrelation)"
display ""
display "Current result: p-value = 0.0009 < 0.05"
display "→ We REJECT H0: Residuals exhibit significant autocorrelation"
display "→ ARMA(2,1) may not fully capture all dynamics in the data"


(note: time series has 1949 gaps)

Portmanteau test for white noise
---------------------------------------
 Portmanteau (Q) statistic =    45.8134
 Prob > chi2(20)           =     0.0009



H0: No autocorrelation in residuals up to lag 20



Interpretation:

- If p-value > 0.05, we fail to reject H0 (residuals are white noise)

- If p-value <= 0.05, we reject H0 (residuals show autocorrelation)



Current result: p-value = 0.0009 < 0.05

→ We REJECT H0: Residuals exhibit significant autocorrelation

→ ARMA(2,1) may not fully capture all dynamics in the data


#### Interpreting the Ljung-Box Test Results

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**⚠️ Test Result Interpretation**

The Ljung-Box Q-test results show:

- **Q statistic** = 45.8134
- **p-value** = 0.0009 (< 0.05)
- **Conclusion**: We **REJECT** H₀

**What this means:**

1. **Residuals are NOT white noise**: The test detects significant autocorrelation in the residuals at lags up to 20
2. **Model misspecification**: ARMA(2,1) may not fully capture all dynamics in the log-VIX series
3. **Possible explanations**:
   - Missing higher-order AR/MA terms
   - Presence of conditional heteroskedasticity (volatility clustering) that ARMA cannot capture
   - Non-linear dynamics requiring more sophisticated models
   - Effects of the 1949 gaps in the time series on the test

**Note on gaps**: The presence of 1949 gaps in the time series may affect the validity of the Ljung-Box test, as it assumes a complete time series. The missing observations could potentially distort the autocorrelation patterns.

</div>

<div style="background: #fef3c7; border-left: 4px solid #f59e0b; padding: 18px; border-radius: 8px; margin: 20px 0;">

**🔍 Next Steps**

While the ARMA(2,1) model may be adequate for short-term forecasting (as shown in the out-of-sample evaluation), the Ljung-Box test suggests that:

- For a complete diagnostic picture, examine ACF/PACF plots of residuals more carefully
- Consider exploring GARCH models (Session 2) to capture volatility clustering
- The model still provides useful forecasts despite this diagnostic failure, which is common in financial time series

</div>


### 2.7 Fitted vs Empirical ACF/PACF Comparison

#### Model Validation Through ACF Matching

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**The Theoretical Foundation**

An ARMA model implies a **specific theoretical ACF pattern**. For ARMA(2,1):

$$
\rho(k) = \text{Cov}(Y_t, Y_{t-k}) / \text{Var}(Y_t)
$$

is determined by the estimated parameters $\hat{\phi}_1, \hat{\phi}_2, \hat{\theta}_1$.

**Validation Logic**:
- Compute ACF from **fitted values** generated by the model
- Compare with ACF from **empirical data**
- Close match → model captures true autocorrelation structure
- Systematic deviations → model may be misspecified

</div>

#### What to Examine

1. **Overall shape**: Do both ACFs decay at similar rates?
2. **Key lags**: Are lag-1, lag-2 autocorrelations similar?
3. **Tail behavior**: Does the fitted ACF converge to zero smoothly like the empirical ACF?

<div style="background: #f0fdf4; border-left: 4px solid #10b981; padding: 16px; border-radius: 6px; margin: 16px 0;">

**Expected Outcome**: For a correctly specified ARMA(2,1), the fitted ACF should closely track the empirical ACF across all lags, providing visual confirmation that the model has successfully replicated the data's correlation structure.

</div>

In [44]:
* Compare fitted ACF from ARMA(2,1) with empirical ACF
* First, compute empirical ACF values
quietly corrgram lgvix, lags(40)  // Compute correlogram for log-VIX up to 40 lags

* Get fitted values and compute their ACF
predict fitted_lgvix, xb  // Generate linear prediction (fitted values) from ARMA model
quietly corrgram fitted_lgvix, lags(40)  // Compute correlogram for fitted values up to 40 lags

display "The fitted ARMA(2,1) model should closely match the empirical ACF pattern"
display "Any systematic deviations suggest model misspecification"





The fitted ARMA(2,1) model should closely match the empirical ACF pattern

Any systematic deviations suggest model misspecification


### 2.8 Out-of-Sample Forecasting and Evaluation

#### The Ultimate Test: Predictive Performance

<div style="background: linear-gradient(135deg, #fff5f5 0%, #ffe5e5 100%); border-left: 4px solid #dc2626; padding: 18px; border-radius: 8px; margin: 20px 0;">

**From Fitting to Forecasting**

While in-sample diagnostics validate model specification, **out-of-sample forecasting** tests the model's practical value:
- Can the model predict future VIX values accurately?
- How does forecast accuracy deteriorate as the horizon increases?
- Are confidence intervals properly calibrated?

</div>

#### Dynamic Forecasting Framework

We employ **dynamic multi-step forecasting**:

$$
\hat{Y}_{T+h|T} = \phi_1 \hat{Y}_{T+h-1|T} + \phi_2 \hat{Y}_{T+h-2|T} + \theta_1 \hat{\varepsilon}_{T+h-1|T}
$$

where:
- $\hat{Y}_{T+h|T}$ is the $h$-step ahead forecast made at time $T$
- Past forecasts replace actual values as horizon extends
- Forecast errors compound, widening confidence intervals

**Confidence Intervals**:
$$
\text{95\% CI: } \quad \hat{Y}_{T+h|T} \pm 1.96 \times \sqrt{\text{MSE}(h)}
$$

#### Performance Metrics

<div style="background: #f0f9ff; border-left: 4px solid #3b82f6; padding: 18px; border-radius: 8px; margin: 20px 0;">

**Forecast Accuracy Measures**

1. **Mean Error (ME)**: $\frac{1}{n}\sum (Y_t - \hat{Y}_t)$
   - Measures bias (should be ≈ 0 for unbiased forecasts)

2. **Mean Absolute Error (MAE)**: $\frac{1}{n}\sum |Y_t - \hat{Y}_t|$
   - Average magnitude of errors (lower = better)

3. **Root Mean Squared Error (RMSE)**: $\sqrt{\frac{1}{n}\sum (Y_t - \hat{Y}_t)^2}$
   - Penalizes large errors more heavily
   - Standard metric for comparing forecast methods

</div>

#### Interpretation Guidelines

**Good forecasts exhibit**:
- ME close to zero (unbiased)
- Low MAE and RMSE relative to series volatility
- Most actual values within 95% confidence bands
- Errors that don't exhibit systematic patterns (no autocorrelation)

In [45]:
* Create a sample split for out-of-sample forecasting
* Use data up to end of 2014 for estimation, forecast into Jan 2015

* First, identify the last observation in 2014
summarize date  // Display summary of date variable
gen year = year(date)  // Extract year from date variable
gen month = month(date)  // Extract month from date variable

* Save last date in dataset for reference
quietly summarize date  // Compute date summary statistics without output
local last_date = r(max)  // Store maximum date value in local macro

* Re-estimate ARMA(2,1) on full sample for forecasting
arima lgvix, ar(1/2) ma(1) vce(robust)  // Re-estimate ARMA(2,1) model for forecasting

* Generate forecasts for next 20 periods (approximately 1 month of trading days)
predict lgvix_forecast, dynamic(td(01jan2015))  // Generate dynamic forecasts starting Jan 1, 2015
predict lgvix_se, mse  // Generate mean squared error estimates for confidence intervals

* Convert forecasts back to VIX level
gen vix_forecast = exp(lgvix_forecast)  // Transform log-VIX forecasts back to VIX level

* Calculate confidence intervals
gen lgvix_lb = lgvix_forecast - 1.96*sqrt(lgvix_se)  // Calculate lower bound of 95% CI in log scale
gen lgvix_ub = lgvix_forecast + 1.96*sqrt(lgvix_se)  // Calculate upper bound of 95% CI in log scale
gen vix_lb = exp(lgvix_lb)  // Transform lower bound to VIX level
gen vix_ub = exp(lgvix_ub)  // Transform upper bound to VIX level

* Display forecast statistics
display "Out-of-Sample Forecast Summary"
display "==============================="
summarize vix vix_forecast if date >= td(01jan2015) & date <= td(31jan2015)



    Variable |        Obs        Mean    Std. dev.       Min        Max
-------------+---------------------------------------------------------
        date |      9,059    17520.19    3791.214      10959      24058





(note:  insufficient memory or observations to estimate usual
starting values [2])

Number of gaps in sample = 1949
(note: filtering over missing observations)

(setting optimization to BHHH)
Iteration 0:   log pseudolikelihood = -3148.9011  (not concave)
Iteration 1:   log pseudolikelihood =  4513.0107  
Iteration 2:   log pseudolikelihood =  5600.0015  
Iteration 3:   log pseudolikelihood =  6107.6875  
Iteration 4:   log pseudolikelihood =  10529.408  
(switching optimization to BFGS)
Iteration 5:   log pseudolikelihood =  10763.231  
Iteration 6:   log pseudolikelihood =  11352.161  
Iteration 7:   log pseudolikelihood =  11371.627  
Iteration 8:   log pseudolikelihood =  11386.643  
Iteration 9:   log pseudolikelihood =  11387.939  
Iteration 10:  log pseudolike

In [46]:
* Plot forecasts vs actual values for January 2015
twoway (line vix date if date >= td(01dec2014) & date <= td(31jan2015), lcolor(blue) lwidth(medium)) ///
       (line vix_forecast date if date >= td(01dec2014) & date <= td(31jan2015), lcolor(red) lpattern(dash) lwidth(medium)) ///
       (rarea vix_lb vix_ub date if date >= td(01jan2015) & date <= td(31jan2015), fcolor(red%20) lwidth(none)), ///
    title("VIX Out-of-Sample Forecasts vs Actual", size(medium)) ///
    subtitle("ARMA(2,1) Model - January 2015") ///
    ytitle("VIX Index") ///
    xtitle("Date") ///
    tlabel(01dec2014(15)31jan2015, format(%tdCY)) ///
    legend(order(1 "Actual VIX" 2 "Forecast" 3 "95% CI") rows(1)) ///
    scheme(s2color)
graph export "../../output/vix_forecast.png", replace width(1200)



file /Users/jesusvillotamiranda/.stata_kernel_cache/graph23.svg saved as SVG
    format
file /Users/jesusvillotamiranda/.stata_kernel_cache/graph23.pdf saved as PDF
    format

(file ../../output/vix_forecast.png not found)
failed to export to the specified format.


r(198);
r(198);


In [ ]:
* Calculate forecast accuracy metrics
gen forecast_error = vix - vix_forecast if date >= td(01jan2015) & date <= td(31jan2015)  // Compute forecast errors for January 2015
gen abs_error = abs(forecast_error)  // Calculate absolute value of forecast errors
gen squared_error = forecast_error^2  // Calculate squared forecast errors

quietly summarize forecast_error if date >= td(01jan2015) & date <= td(31jan2015)  // Compute mean error statistics
local me = r(mean)  // Store mean error in local macro
quietly summarize abs_error if date >= td(01jan2015) & date <= td(31jan2015)  // Compute mean absolute error statistics
local mae = r(mean)  // Store mean absolute error in local macro
quietly summarize squared_error if date >= td(01jan2015) & date <= td(31jan2015)  // Compute mean squared error statistics
local mse = r(mean)  // Store mean squared error in local macro
local rmse = sqrt(`mse')  // Calculate root mean squared error from MSE

display ""
display "Forecast Accuracy Metrics (January 2015):"
display "=========================================="
display "Mean Error (ME):           " %8.4f `me'
display "Mean Absolute Error (MAE): " %8.4f `mae'
display "Root Mean Squared Error:   " %8.4f `rmse'
display ""
display "Interpretation:"
display "- ME close to 0 suggests unbiased forecasts"
display "- Lower MAE and RMSE indicate better forecast accuracy"

<div style="max-width: 700px; margin: 20px auto 30px; padding: 20px; text-align: center; border-bottom: 2px solid #e5e7eb;">
  <a href="1_1_simul_univariate.ipynb" style="display: inline-block; padding: 10px 24px; background: #f9fafb; color: #38549c; text-decoration: none; border-radius: 8px; font-weight: 600; font-size: 14px; border: 1px solid #e5e7eb; margin-right: 12px;">
    ← Part 1 - Univariate Time Series Model Simulations
  </a>
</div>
